### ISSUE TITLE
imgpkg pull removes duplicate image references 

### ISSUE THREAD

**Issue by matthewmcnew** [carvel-dev/imgpkg#176]
**What steps did you take:**
* Create imgpkg bundle with an `.imgpkg/images.yml` with duplicate image references. Example included below:
```
---
apiVersion: imgpkg.carvel.dev/v1alpha1
kind: ImagesLock
images:
- image: my-repo.io/sample@sha256:3f8342b534e3e308188c3d0683c02c941c407a1ddacb086425499ed9cf0888e9
  annotations:
    kbld.carvel.dev/id: setup-ca-certs
- image: my-repo.io/sample@sha256:3f8342b534e3e308188c3d0683c02c941c407a1ddacb086425499ed9cf0888e9
  annotations:
    kbld.carvel.dev/id: my-repo.io/sample@sha256:3f8342b534e3e308188c3d0683c02c941c407a1ddacb086425499ed9cf0888e9
```
* Utilize `imgpkg copy` to relocate the imgpkg to a different registry
* Use `imgpkg pull` to pull the bundle from the new location. 

**What happened:**
The pulled and rewritten imgpkg bundle only includes one reference to the image originally duplicated in images.yml
```
---
apiVersion: imgpkg.carvel.dev/v1alpha1
images:
kind: ImagesLock
- annotations:
    kbld.carvel.dev/id: my-repo.io/sample@sha256:3f8342b534e3e308188c3d0683c02c941c407a1ddacb086425499ed9cf0888e9
  image: my-repo.io/sample@sha256:3f8342b534e3e308188c3d0683c02c941c407a1ddacb086425499ed9cf0888e9
```

**What did you expect:**
The pulled images.yml was expected to include two references to the relocated images. 

**Environment:**

- imgpkg version (use `imgpkg --version`): 0.9+

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by joaopapereira** [comment]
We are able to reproduce this issue.

`imgpkg` is indeed squashing the ImagesLock file when there are 2 or more images with the same field Image. This is a scenario that can happen when a user is integrating multiple manifests where the same image can be referenced with different strings or tag vs sha.

Going to accept this bug.

**Comment by joaopapereira** [comment]
Fixed in version [0.12.0](https://github.com/vmware-tanzu/carvel-imgpkg/releases/tag/v0.12.0)
